pkg/controller/pxc/backup.go

```diff
@@ -26,6 +28,31 @@ func (r *ReconcilePerconaXtraDBCluster) reconcileBackups(cr *api.PerconaXtraDBCl
 	if cr.Spec.Backup != nil {
 		bcpObj := backup.New(cr)
 
+		if cr.Spec.Backup.EnablePITR {
+			var binlogCollector appsv1.Deployment
+			binlogCollectorName := cr.Name + "-bl-collector"
+
+			err = r.client.Get(context.TODO(), types.NamespacedName{Name: binlogCollectorName, Namespace: cr.Namespace}, &binlogCollector)
+			if err != nil && k8serrors.IsNotFound(err) {
+				binlogCollector, err := deployment.GetBinlogCollectorDeployment(cr)
+				if err != nil {
+					return fmt.Errorf("get binlog collector deployment for cluster '%s': %v", cr.Name, err)
+				}
+				err = r.client.Create(context.TODO(), &binlogCollector)
+				if err != nil {
+					return fmt.Errorf("create binlog collector deployment for cluster '%s': %v", cr.Name, err)
+				}
+			} else if err != nil {
+				return fmt.Errorf("get binlogCollector '%s': %v", binlogCollectorName, err)
+			} else {
+				err = r.client.Update(context.TODO(), &binlogCollector)
```

Comment:
`binlogCollector` is empty here

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/622/commits/aa1d6dc5ff0363bf607312ebc1dc9de918f418d2

Reply:
the current approach can recreate Deployment on each reconcile loop.
in the future, we should have a test which checks if it is not a true